packages/components/src/components/post-tooltip/post-tooltip.tsx

```diff
@@ -0,0 +1,277 @@
+import { Component, Element, h, Host, Method, Prop, State, Watch } from '@stencil/core';
+import {
+  arrow,
+  autoUpdate,
+  computePosition,
+  flip,
+  inline,
+  offset,
+  Placement,
+  shift,
+} from '@floating-ui/dom';
+import isFocusable from 'ally.js/esm/is/focusable';
+
+// Polyfill for popovers, can be removed when https://caniuse.com/?search=popover is green
+import '@oddbird/popover-polyfill';
+
+// Patch for long press on touch devices
+import 'long-press-event';
+
+import { version } from '../../../package.json';
+import { checkType } from '../../utils';
+
+const SIDE_MAP = {
+  top: 'bottom',
+  right: 'left',
+  bottom: 'top',
+  left: 'right',
+};
+
+export type BackgroundColor = 'primary' | 'brand-yellow';
+
+interface PopoverElement {
+  showPopover: () => void;
+  hidePopover: () => void;
+  togglePopover: (force?: boolean) => boolean;
+}
+
+@Component({
+  tag: 'post-tooltip',
+  styleUrl: 'post-tooltip.scss',
+  shadow: true,
+})
+export class PostTooltip {
+  private tooltipRef: HTMLDivElement & PopoverElement;
+  private arrowRef: HTMLElement;
+  private clearAutoUpdate: () => void;
+  private readonly localShowTooltip: (e: Event) => Promise<void>;
+  private readonly localHideTooltip: () => Promise<void>;
+  private readonly localToggleTooltip: () => Promise<void>;
+  private eventTarget: Element;
+
+  @Element() host: HTMLPostTooltipElement;
+
+  @State() tooltipClasses: string;
+
+  /**
+   * Defines the background color of the tooltip.
+   * Choose the one that provides the best contrast in your scenario.
+   */
+  @Prop() readonly backgroundColor?: BackgroundColor = 'primary';
+
+  /**
+   * Defines the placement of the tooltip according to the floating-ui options available at https://floating-ui.com/docs/computePosition#placement.
+   * Tooltips are automatically flipped to the opposite side if there is not enough available space and are shifted
+   * towards the viewport if they would overlap edge boundaries.
+   */
+  @Prop() readonly placement?: Placement = 'top';
+
+  @Watch('backgroundColor')
+  validateBackgroundColor(newValue = this.backgroundColor) {
+    checkType(newValue, 'string', 'The post-tooltip "background-color" prop should be a string.');
+
+    if (newValue === 'brand-yellow') {
```

Comment:
I would just go for "yellow" as both the SASS variable and the background class are simply called "yellow" as well.

```suggestion
    if (newValue === 'yellow') {
```